File: Cs/Acme/Application/Command/Handler/CreateUserHandler.cs
using Cs.Acme.Domain;
using Cs.Acme.Domain.Model;
using Cs.SharedKernel.Application.Command;
using Cs.SharedKernel.Application.Command.Handler;
using System;

namespace Cs.Acme.Application.Command.Handler
{
    public class CreateUserHandler : IHandler
    {
        private readonly IUsers _users;

        public CreateUserHandler(IUsers users)
        {
            _users = users;
        }

        public void Handle(ICommand command)
        {
            var createUserCommand = (CreateUserCommand) command;
            if (_users.Exists(createUserCommand.Email))
            {
                throw new Exception("Current user exists");
            }
            
            var user = new User(
                createUserCommand.Email,
                createUserCommand.Password,
                createUserCommand.FirstName,
                createUserCommand.LastName
            );

            _users.Add(user);
        }
    }
}

File: php8/src/Acme/Application/Command/Handler/CreateUserHandler.php
<?php

declare(strict_types=1);

namespace Acme\Application\Command\Handler;

use Acme\Application\Command\CreateUserCommand;
use Acme\Domain\Model\User;
use Acme\Domain\Users;
use SharedKernel\Application\Command\Command;
use SharedKernel\Application\Command\Handler\Handler;

class CreateUserHandler implements Handler
{
    private Users $users;

    public function __construct(Users $users)
    {
        $this->users = $users;
    }

    /**
     * @throws \Exception
     */
    public function handle(CreateUserCommand|Command $command): void
    {
        if ($this->users->exists($command->email())) {
            throw new \InvalidArgumentException('Current user exists');
        }

        $user = new User($command->email(), $command->password(), $command->firstName(), $command->lastName());

        $this->users->add($user);
    }
}
